YarnSpinner/yarn_spinner.proto

```diff
@@ -3,131 +3,147 @@ package Yarn;
+        // opA = string: string ID for option to add
+        ADD_OPTION = 4; 
+        
+        // Presents the current list of options to the client, then clears
+        // the list. The most recently selected option will be on the top
+        // of the stack when execution resumes.
+        // No operands.
+        SHOW_OPTIONS = 5; 
+        
+        // Pushes a string onto the stack.
+        // opA = string: the string to push to the stack.
+        PUSH_STRING = 6; 
+        
+        // Pushes a floating point number onto the stack.
+        // opA = float: number to push to stack
+        PUSH_FLOAT = 7; 
+        
+        // Pushes a boolean onto the stack.
+        // opA = bool: the bool to push to stack
+        PUSH_BOOL = 8; 
+        
+        // Pushes a null value onto the stack.
+        // No operands.
+        PUSH_NULL = 9; 
+        
+        // Jumps to the named position in the the node, if the top of the
+        // stack is not null, zero or false.
+        // opA = string: label name 
+        JUMP_IF_FALSE = 10; 
+        
+        // Discards top of stack.
+        // No operands.
+        POP = 11; 
+        
+        // Calls a function in the client. Pops as many arguments as the
+        // client indicates the function receives, and the result (if any)
+        // is pushed to the stack.
+        
+        // opA = string: name of the function
+        CALL_FUNC = 12; 
+        
+        // Pushes the contents of a variable onto the stack.
+        // opA = name of variable 
+        PUSH_VARIABLE = 13; 
+        
+        // Stores the contents of the top of the stack in the named
+        // variable. 
+        // opA = name of variable
+        STORE_VARIABLE = 14; 
+        
+        // Stops execution of the program.
+        // No operands.
+        STOP = 15; 
+        
+        // Run the node whose name is at the top of the stack.
+        // No operands.
+        RUN_NODE = 16; 
     }
 }
 
 // A value used by an Instruction.
 message Operand {
     oneof value {
-		
-		// A string.
-		string string_value = 1;
-		
-		// A boolean (true or false).
-		bool bool_value = 2;
+        // A string.
+        string string_value = 1;
+        
+        // A boolean (true or false).
+        bool bool_value = 2;
 
-		// A floating point number.
-		float float_value = 3;
+        // A floating point number.
+        float float_value = 3;
     }
 }
+
+// Virtual Machine states currentOptions KeyValuePair
+message VMStateOption {
+    string key = 1;
+    string value = 2;
+}
+
+// Serializable encloser for Yarn Value
+message VMStateValue {
+    int32 type = 1;
+    string value = 2;
+}
+
+// Virtual Machine state
+message VMState {
+    string currentNodeName = 1;
+    int32 programCounter = 2; //as per protobuf documentation: For numeric types, the default value is zero.
+    repeated VMStateOption currentOptions = 3;
+    repeated VMStateValue stack = 4;
+}
```

Comment:
I don't feel that this should go inside the same proto file.
The rest of the messages in this file are for what become assets, the VM state isn't an asset.

Would you mind pulling this out into its own definition file?

Reply:
Ok thank you, I wasn't entirely sure if I should make new proto for those. Will move it.